packages/core/src/app/checkout/Checkout.tsx

```diff
@@ -423,30 +424,44 @@ class Checkout extends Component<
     }
 
     private renderShippingStep(step: CheckoutStepStatus): ReactNode {
-        const { hasCartChanged, cart, consignments = [] } = this.props;
+        const { hasCartChanged, cart, consignments = [], isNewMultiShippingUIEnabled } = this.props;
 
         const { isBillingSameAsShipping, isMultiShippingMode } = this.state;
 
         if (!cart) {
             return;
         }
 
+        const isRenderMultiStaticConsignment = isNewMultiShippingUIEnabled && consignments.length > 1;
```

Comment:
Can we use `isMultiShippingMode` instead of checking consignments.length?

Reply:
Also what happens in case when someone chooses multi-shipping mode but only create 1 consignment, does that mean it is single shipping?

Reply:
>Also what happens in case when someone chooses multi-shipping mode but only create 1 consignment, does that mean it is single shipping?

Choosing multi-shipping mode will update `isMultiShippingMode` as `true`, and the UI will stay in the multi-shipping mode until clicking the switch button.

Reply:
Also, new changes should be behind the experiment in order to not break any existing multi-shipping (CSS) customizations merchants may have.

Reply:
Thanks for suggestion. Looks like `isMultiShippingMode` already check the case only created 1 consignment, and will return false. I will change to `isNewMultiShippingUIEnabled && isMultiShippingMode`

https://github.com/bigcommerce/checkout-js/blob/CHECKOUT-8600/packages/core/src/app/shipping/isUsingMultiShipping.ts#L10-L21